tests/test_ui.py

```diff
@@ -0,0 +1,597 @@
+        ), 'first result should be active'
+
+        search_outer_input.send_keys(Keys.DOWN)
+        assert results[1] == selenium.find_element_by_css_selector(
+            '.search__result__single.active'
+        ), 'second result should be active'
+
+        search_outer_input.send_keys(Keys.UP)
+        search_outer_input.send_keys(Keys.UP)
+        assert results[-1] == selenium.find_element_by_css_selector(
+            '.search__result__single.active'
+        ), 'last result should be active'
+
+        search_outer_input.send_keys(Keys.DOWN)
+        assert results[0] == selenium.find_element_by_css_selector(
+            '.search__result__single.active'
+        ), 'first result should be active'
+
+
+@pytest.mark.sphinx(srcdir=TEST_DOCS_SRC)
+def test_enter_button_on_input_field_when_no_result_active(selenium, app, status, warning):
+    """
+    Test if pressing Enter on search field takes the user to search page.
+
+    User will be redirected to the search page only if no result is active.
+    A search result becomes active if the user reaches to it via ArrowUp and
+    ArrowDown buttons.
+    """
+    app.build()
+    path = app.outdir / 'index.html'
+
+    # to test this, we need to override the $.ajax function
+    ajax_func = '''
+        <script>
+            $.ajax = function(params) {
+                return params.error(
+                    { },
+                    'error',
+                    'Dummy Error.'
+                )
+            }
+        </script>
+    '''
+
+    injected_script = SCRIPT_TAG + ajax_func
+
+    with InjectJsManager(path, injected_script) as _:
+        selenium.get(f'file://{path}')
+        open_search_modal(selenium)
+
+        search_outer_input = selenium.find_element_by_class_name(
+            'search__outer__input'
+        )
+        search_outer_input.send_keys('i am searching')
+        search_outer_input.send_keys(Keys.ENTER)
+
+        WebDriverWait(selenium, 10).until(
+            EC.url_matches(app.outdir / 'search.html')
+        )
+
+        # enter button should redirect the user to search page
+        assert (
+            'search.html' in selenium.current_url
+        ), 'search.html must be in the url of the page'
+
+        assert (
+            'Search' in selenium.title
+        ), '"Search" must be in the title of the page'
+
+
+@pytest.mark.sphinx(srcdir=TEST_DOCS_SRC)
+def test_position_search_modal(selenium, app, status, warning):
+    """Test if the search modal is in the middle of the page."""
+    app.build()
+    path = app.outdir / 'index.html'
+
+    with InjectJsManager(path, SCRIPT_TAG) as _:
+        selenium.get(f'file://{path}')
+        open_search_modal(selenium)
+
+        search_outer_wrapper = selenium.find_element_by_class_name(
+            'search__outer__wrapper'
+        )
+        search_outer = selenium.find_element_by_class_name(
+            'search__outer'
+        )
+
+        window_sizes = [
```

Comment:
We should probably also make sure that the model isn't broken on a mobile-sized viewport. It doesn't look like it is, but it's probably worth a test.

Reply:
I have added other sizes as well.